apps/rush-lib/src/api/LastInstallFlag.ts

```diff
@@ -27,17 +73,56 @@ export class LastInstallFlag {
 
   /**
    * Returns true if the file exists and the contents match the current state
+   * @param abortOnInvalid - If the current state is not equal to the previous
+   * state, and an the current state causes an error, then throw an exception
+   * with a friendly message
    */
-  public isValid(): boolean {
+  public isValid(abortOnInvalid: boolean = false): boolean {
     if (!FileSystem.exists(this._path)) {
       return false;
     }
+    let contents: JsonObject;
     try {
-      const contents: JsonObject = JsonFile.load(this._path);
-      return _.isEqual(contents, this._state);
-    } catch (error) {
+      contents = JsonFile.load(this._path);
+    } catch (err) {
+      return false;
+    } 
+    if (!_.isEqual(contents, this._state)) {
+      if (abortOnInvalid) {
+        const flagError: LastInstallFlagError | void = this._stateHasError(contents);
+        if (flagError) {
+          throw flagError;
+        }
+      }
       return false;
     }
+    return true;
+  }
+
+  /**
+   * Determine if the new LastInstallFlag state should generate an error
+   * @param oldState - The LastInstallFlag contents read from the disk
+   */
+  private _stateHasError(oldState: JsonObject): LastInstallFlagError | void {
+    const pkgManager: PackageManagerName = this._state.packageManager;
+    switch (pkgManager) {
+      case 'pnpm': {
+        if (
+          ( // Only throw an error if the package manager hasn't changed from pnpm
+            oldState.packageManager === pkgManager
+          ) && ( // Throw if the store path changed
+            oldState.storePath &&
```

Comment:
@iclanton I just realized that this should be `oldState.storePath !== undefined`.

If the default global path is used, the value here will be "".  So when the store value is changed to a different location, this condition will not catch the error state, and it will fall through to PNPM, which will throw an exception instead.

Additionally, this check has to be here, because if somebody is changing package managers, then this property will be `undefined`, which is **not** an error state.

This is something the tests should have caught, but I don't think I checked for "" -> "/some/path"... just "/some/path" !== "/some/other/path" :man_facepalming: